crates/shared/src/order_validation.rs

```diff
@@ -434,6 +435,10 @@ impl OrderValidating for OrderValidator {
             if let TokenQuality::Bad { reason } = self
                 .bad_token_detector
                 .detect(token)
+                .instrument(tracing::info_span!(
```

Comment:
I'm a bit surprised by all these instrumented futures and think these could lead to duplicated tracing spans. Wouldn't it be sufficient to instrument only the `inner.detect()` calls inside the `CachingDetector`?

Reply:
This would assume we always use a `CachingDetector` in our estimating chain (which is true now but might change when and we don't want the observability to change in that case). I'm fine with this, but I thought the callsite that creates the outmost

Reply:
Actually, I found `InstrumentedBadTokenDetector` which lends itself perfectly for this.